llm_on_ray/inference/chat_template_process.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright 2023 The LLM-on-Ray Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import List
+
+from llm_on_ray.inference.api_openai_backend.openai_protocol import ChatMessage
+
+
+class ChatTemplatePreprocess:
+    def __init__(self, predictor) -> None:
+        self.predictor = predictor
+
+    def get_prompt(self, input: List, is_mllm=False):
+        """Generate response based on input."""
+        self.predictor.tokenizer.chat_template = (
+            self.predictor.infer_conf.model_description.chat_template
+            or self.predictor.tokenizer.chat_template
+            or self.predictor.infer_conf.model_description.default_chat_template
+        )
+
+        if isinstance(input, list) and input and isinstance(input[0], (ChatMessage, dict)):
+            messages = (
+                [dict(chat_message) for chat_message in input]
+                if isinstance(input[0], ChatMessage)
+                else input
+            )
+            prompt = self.predictor.tokenizer.apply_chat_template(
```

Comment:
This line will also run if is_mllm is true, which is not needed. You can put it in an `else`.